sklearn/ensemble/_bagging.py

```diff
@@ -354,9 +342,11 @@ def fit(self, X, y, sample_weight=None, **fit_params):
             regression).
 
         sample_weight : array-like of shape (n_samples,), default=None
-            Sample weights. If None, then samples are equally weighted.
-            Note that this is supported only if the base estimator supports
-            sample weighting.
+            Sample weights. If None, then samples are equally weighted. Used as
+            probabilities to sample the training set. Note that the expected
+            frequency semantics for the `sample_weight` parameter are only
+            fulfilled when sampling with replacement `bootstrap`=True.
```

Comment:
```suggestion
            fulfilled when sampling with replacement `bootstrap=True`.
```

Hopefully, this was the cause of the following sphinx warning that currently breaks the CI although I am not really sure:

```
sklearn/ensemble/_bagging.py:docstring of sklearn.ensemble._bagging.BaseBagging.fit:16: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils] 
```